dev-tools/mage/build.go

```diff
@@ -67,7 +67,7 @@ func DefaultBuildArgs() BuildArgs {
 
 	if DevBuild {
 		// Disable optimizations (-N) and inlining (-l) for debugging.
-		args.ExtraFlags = append(args.ExtraFlags, `-gcflags`, `"all=-N -l"`)
+		args.ExtraFlags = append(args.ExtraFlags, `-gcflags=all=-N -l`)
```

Comment:
these flags do not work on masOS

Reply:
you are right, I have testest it locally on a Intel MacBook and it works
```
DEV=1 MAGEFILE_VERBOSE=true mage build
Running target: Build
>> build: Building heartbeat
exec: git "rev-parse" "HEAD"
Adding build environment vars: map[CGO_ENABLED:1]
exec: go "build" "-o" "heartbeat" "-buildmode" "pie" "-gcflags=all=-N -l" "-ldflags" "-X github.com/elastic/beats/v7/libbeat/version.buildTime=2022-07-08T15:52:38Z -X github.com/elastic/beats/v7/libbeat/version.commit=fc9849d755f9432800d1a17e031bb81e6810fcba"
```

sorry for the noise, I have tested too many things these two days that are difficult to remember 'em all correctly